### PR TITLE
Eventslist(upcomingevents)

### PR DESCRIPTION
**Pull request by moatazalaa14**

### CODE REVIEW

File: src/modules/events/components/eventsListPage/components/upComingEvents/index.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            <button className={styles.upcomingEvent__btnContainer__btn}>
+              Register Now
+            </button>
+          </div>
+        </div>
+      );
+    });
+    return (
+      <div className={styles.eventsList}>
+        <h1 className={styles.eventsList__heading}>UpComing Events</h1>
```
Comment (dvmhmdsd) [inline src/modules/events/components/eventsListPage/components/upComingEvents/index.js]: The heading is `Our Events`

File: src/modules/events/components/eventsListPage/components/upComingEvents/style.module.css [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    justify-items: center;
+
+}
+.upcomingEvent{
+    display: grid;
+    grid-template-columns: 1fr 1fr;
+    border: white solid 1px;
+    background-image:linear-gradient(rgba(0,0,0,0.7),rgba(0,0,0,0.7)), url(../../../../../../assets/images/event.jpg);
+    height: 50vh;
+    width: 80vw;
```
Comment (dvmhmdsd) [inline src/modules/events/components/eventsListPage/components/upComingEvents/style.module.css]: Why did u add a fixed width and height ???

File: src/modules/events/components/eventsListPage/components/upComingEvents/style.module.css [hunk 3]
```diff
@@ -0,0 +1,39 @@
+.eventsList{
```
Comment (dvmhmdsd) [inline src/modules/events/components/eventsListPage/components/upComingEvents/style.module.css]: Use `BEM` --> `events-list`

File: src/modules/events/components/eventsListPage/components/upComingEvents/style.module.css [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    display: grid;
+    grid-template-columns: 1fr;
+    justify-items: center;
+
+}
+.upcomingEvent{
```
Comment (dvmhmdsd) [inline src/modules/events/components/eventsListPage/components/upComingEvents/style.module.css]: Use `BEM` --> `upcoming-event`, and please use a descriptive class names.

**Review by dvmhmdsd — CHANGES_REQUESTED** [review]
1. Don't rename the components, leave it `eventsListPage` as it's and don't modify the structure of the folders please.

2. Use the `EventCard` component and pass props to it.

3. Make the events data `async` by getting it from the  `events.json` in the public folder (see bestMembers component for illustration).

**Review by dvmhmdsd — CHANGES_REQUESTED** [review]
- There should be 3 cards in the row on desktop, 2 on tablet and 1 on mobile.

- Don't add any style for the card and use the `EventCard` component.

- Don't add any bg-image please.

**Review by dvmhmdsd — CHANGES_REQUESTED** [review]
1. Use the `eventCard` component and pass the data of the event to it.
2. You didn't do anything for making the page responsive.

**Comment by dvmhmdsd** [comment]
- Add the title of the page `Our events`, and a heading for upcoming event and another heading for past events. Also please use the  `eventCard` Component.

**Comment by dvmhmdsd** [comment]
You should do the following:

1. Request the events.
2. Update the state with the response.
3. In JSX:
   - Loop across the data in the state.
   - Pass the single object to `eventCard` Component

--> Consider this code for illustration:
```js
events.map(singleEvent => {
   return (
      <div className="custom-class">
         <EventCard event={singleEvent} />
     </div>
   )
});
```

File: src/modules/events/components/eventsListPage/index.js [hunk 5]
```diff
@@ -1,7 +1,58 @@
...
+                      <div className={styles["upcoming-events"]}>
+                        <EventCard event={event} />
+                      </div>
+                    );
+                  } else {
+                    return null;
+                  }
+                })
+              : null}
+            <h1 className={styles[`past-heading`]}>Past events</h1>
```
Comment (IEEE-AlAzhar) [inline src/modules/events/components/eventsListPage/index.js]: There should be just 1 `h1` on the page.

File: src/modules/events/components/eventsListPage/index.js [hunk 6]
```diff
@@ -1,7 +1,58 @@
...
+            {this.state.eventsList
+              ? this.state.eventsList.map((event) => {
+                  if (event.status !== "upcoming") {
+                    return (
+                      <div>
```
Comment (IEEE-AlAzhar) [inline src/modules/events/components/eventsListPage/index.js]: This div is not appropriate.

**Review by IEEE-AlAzhar — CHANGES_REQUESTED** [review]

File: src/modules/events/components/eventsListPage/index.js [hunk 7]
```diff
@@ -1,7 +1,58 @@
...
+      this.setState({
+        eventsList: res.data,
+      });
+    });
+  }
   render() {
-    return <h1> Events list page is ready ! </h1>;
+    return (
+      <>
+        {this.state.eventsList ? console.log(this.state.eventsList) : null}
```
Comment (IEEE-AlAzhar) [inline src/modules/events/components/eventsListPage/index.js]: Console !!!

File: src/modules/events/components/eventsListPage/index.js [hunk 8]
```diff
@@ -1,7 +1,58 @@
...
+        <Layout>
+          <h1 className={styles[`events-heading`]}>Our Events</h1>
+          <br />
+          <div className={styles[`events-container`]}>
```
Comment (IEEE-AlAzhar) [inline src/modules/events/components/eventsListPage/index.js]: Please use `bootstrap grid` to make the responsive more easier.

**Review by IEEE-AlAzhar — CHANGES_REQUESTED** [review]
- The UI is not appropriate as the cards don't have side spaces.

- The `upcoming` event shouldn't fill the whole space of the row.

- The titles should be:
`h1` Our events
`h2` upcoming events
`h2` past events

**Comment by dvmhmdsd** [comment]
1. Use `bootstrap`'s grid please.
2. If there's multiple upcoming events, they should be beside each other, if there's only 1 upcoming event, it should be alone in the row but don't take the whole row.

File: src/modules/events/components/eventsListPage/index.js [hunk 9]
```diff
@@ -1,7 +1,67 @@
...
+          <div className={styles[`events-list`]}>
+            <div className="container">
+              <h1 className={styles[`events-heading`]}>Our Events</h1>
+              <h2 className={styles[`events-heading`]}>Upcoming Events</h2>
+              <div className="row">
+                {this.state.eventsList ? (
+                  this.state.eventsList.map((event) => {
+                    if (event.status === "upcoming") {
+                      return (
+                        <div className="lg-col-12 md-col-12 sm-col-12 mt-4">
```
Comment (dvmhmdsd) [inline src/modules/events/components/eventsListPage/index.js]: The `key` prop should be here not in eventCard

File: src/modules/events/components/eventsListPage/index.js [hunk 10]
```diff
@@ -1,7 +1,67 @@
...
+                ) : (
+                  <span>Loading ...</span>
+                )}
+                <h2 className={styles[`past-heading`]}>Past Events</h2>
+                <div className="row">
+                  {this.state.eventsList ? (
+                    this.state.eventsList.map((event) => {
+                      if (event.status !== "upcoming") {
+                        return (
+                          <div className="col-lg-4 col-md-6 col-sm-12 mt-4">
```
Comment (dvmhmdsd) [inline src/modules/events/components/eventsListPage/index.js]: Same here...

**Review by dvmhmdsd — CHANGES_REQUESTED** [review]

**Comment by dvmhmdsd** [comment]
The `upcoming` events should be beside each other and the card should take it's normal width and doesn't fill the whole row if it's single.

**Comment by dvmhmdsd** [comment]
Also there should be a padding for the list to get more space above the footer. Also use semantics and divide the page into sections. 
```html
<h1> .. </h1>
<section>
   <h2> .. </h2>
   ....
</section>
<section>
   <h2> .. </h2>
   ....
</section>
```

**Comment by dvmhmdsd** [comment]
Your work is awesome buddy, last edit: Center the upcoming events cards, so that if there's only one upcoming event it should be at the center.

**Pull request merged**